server/endpoints/chat.js

```diff
@@ -11,13 +12,23 @@ const {
   writeResponseChunk,
 } = require("../utils/chats/stream");
 
+let shouldRespond = true;
+
```

Comment:
This variable would be globally controlled in the API, which would break in any multi-user instance since if two chats were running for two different user connections and user one canceled the response, user two's steam would stop at the same time.